starlette/views.py

```diff
@@ -0,0 +1,25 @@
+from functools import update_wrapper
+from starlette.decorators import make_asgi
+from starlette.types import ASGIApp, ASGIInstance, Scope
+
+
+class View:
+    def dispatch(self, scope: Scope) -> ASGIInstance:
+        request_method = scope["method"] if scope["method"] != "HEAD" else "GET"
+        func = getattr(self, request_method.lower(), None)
+        if func is None:
+            raise Exception(
+                f"Method {request_method} is not implemented for this view."
+            )
+        return make_asgi(func)(scope)
+
+    @classmethod
+    def as_view(cls) -> ASGIApp:
+        def view(scope: Scope):
+            self = cls()
+            return self.dispatch(scope)
```

Comment:
Do we need to do the same closure `.as_view()` dance that Django does?

Rather than using `.as_view()` I figure that we just implement the endpoint/view/whatever class as an ASGI application. We're not using multi-threading so I don't think we need the same "put stuff in a closure for safety" dance, but maybe I'm not thinking it through enough.